File: Darts/src/burlton/dartzee/code/screen/stats/player/HoleBreakdownWrapper.java
package burlton.dartzee.code.screen.stats.player;

import burlton.core.code.obj.HashMapCount;

import java.util.ArrayList;

public final class HoleBreakdownWrapper
{
	private HashMapCount<Integer> hmScoreToCount = new HashMapCount<>();
	
	public void increment(int score)
	{
		hmScoreToCount.incrementCount(score);
	}
	public int getCount(int score)
	{
		return hmScoreToCount.getCount(score);
	}
	
	public double getAverage()
	{
		double totalGamesCounted = hmScoreToCount.getTotalCount();
		
		double weightedTotal = 0;
		ArrayList<Integer> scores = hmScoreToCount.getKeysAsVector();
		for (int score : scores)
		{
			int count = hmScoreToCount.getCount(score);
			weightedTotal += (score * count);
		}
		
		return weightedTotal / totalGamesCounted;
	}
	
	public Object[] getAsTableRow(Object holeIdentifier)
	{
		Object[] row = {holeIdentifier, getCount(1), getCount(2), getCount(3), getCount(4), getCount(5), getAverage()};
		return row;
	}
}

File: Darts/src/burlton/dartzee/code/screen/stats/player/HoleBreakdownWrapper.kt
package burlton.dartzee.code.screen.stats.player

import burlton.core.code.obj.HashMapCount

class HoleBreakdownWrapper
{
    private val hmScoreToCount = HashMapCount<Int>()

    fun getAverage(): Double
    {
        val totalGamesCounted = hmScoreToCount.getTotalCount().toDouble()

        var weightedTotal = 0.0
        val scores = hmScoreToCount.getKeysAsVector()
        for (score in scores)
        {
            val count = hmScoreToCount.getCount(score)
            weightedTotal += (score * count).toDouble()
        }

        return weightedTotal / totalGamesCounted
    }

    fun increment(score: Int) = hmScoreToCount.incrementCount(score)
    fun getCount(score: Int) = hmScoreToCount.getCount(score)
    fun getAsTableRow(holeIdentifier: Any) = arrayOf(holeIdentifier, getCount(1), getCount(2), getCount(3), getCount(4), getCount(5), getAverage())
}